game_event/views.py

```diff
@@ -63,7 +66,20 @@ def game_events(request):
         current_date = date.today()
         game_events = game_events.filter(time__gte=current_date)
 
-    return render(request, 'game-events.html', {'game_events': game_events})
+    weather = get_weather_from_api(game_events, urllib.request.urlopen)
+    for game in game_events:
+        game_events_with_weather.append({
+            'id': game.id,
+            'time': game.time,
+            'level_of_game': game.level_of_game,
+            'min_number_of_players': game.min_number_of_players,
+            'max_number_of_players': game.max_number_of_players,
+            'court': game.court,
+            'ball_game': game.ball_game,
+            'weather': weather[game.id]
+        })
```

Comment:
I have some reservations about this implementation:
- We're changing the type of what is being passed along. It used to be whatever `GameEvent.objects.all()` is returning, and now it's a list - I'd prefer the type to be untouched.
- The elements themselves used to be game events and now they are dictionaries - I'd prefer them to stay game events.
- We assume we know all fields a game event should have, but this might not be true, and suppose we add another in the future, it will be hard to figure out that it's not showing because this function is discarding it, because it was unknown when it was written.

All in all, I think we should have the function that gets the weather to replace the events there with events with weather (maybe you'd need to add weather to the model for that - not sure):
```python
# rename to something like get_events_with_weather
def get_weather_from_api(game_events, weather_getter): 
    for game in game_events:
        ...
        ...
        # have the logic here that gets the weather or None and stores it into a variable called weather
        game.weather = weather
    return game_events
```

Reply:
@yftacherzog  We @IdanZimi  changed some stuff in accordance with your comment, but we did skip changing the model due to time schedule.

Reply:
If it allows the assignment of the weather attribute without changing the model then I completely ok with not changing the model.

Reply:
See my new comments on assigning the weather to the game event object directly rather than storing it on a temp list.